src/cls/IPM/CLI.cls

```diff
@@ -438,22 +481,43 @@ ClassMethod %ParseCommandInput(pCommandString As %String, Output pCommandInfo) A
 					Set tAccum = ""
 					Set tState = $$$PREARGUMENT
 				} ElseIf (tState = $$$MODIFIERNAME) {
+					set modifierSubscript = pCommandInfo
 					If $Data(tCommandStructure(pCommandInfo,"modifiers",tAccum)) {
 						Set tModifier = tAccum
 					} ElseIf $Data(tCommandStructure(pCommandInfo,"modifiers",1,tAccum),tAliasFor) {
 						Set tModifier = tAliasFor
+					} ElseIf (($data(pCommandInfo("parameters","actions"),phases)) && (pCommandInfo = "module-action")) || ($listfind(##class(%IPM.Main).#STANDARDPHASES,pCommandInfo)) {
+						
+						// Lifecycle phases can be called directly or through module-action, we need to share modifiers
+						if (pCommandInfo = "module-action") {
+							set phases = $listfromString(phases)
+						} else {
+							set phases = $lb("module-action")
+						}
+						set ptr = 0
+						while $listnext(phases,ptr,currPhase) {
+							if $data(tCommandStructure(currPhase,"modifiers",tAccum)) {
```

Comment:
Am having a bit of an issue following the structure of the subscripted array here. Although slightly out of scope for this work, could you add documentation to this method to elaborate on the structure beyond the top level subscript?